nixos/modules/services/misc/gitlab.nix

```diff
@@ -598,21 +591,33 @@ in {
 
         # JSON is a subset of YAML
         ln -fs ${pkgs.writeText "gitlab.yml" (builtins.toJSON gitlabConfig)} ${cfg.statePath}/config/gitlab.yml
-        ln -fs ${pkgs.writeText "database.yml" databaseYml} ${cfg.statePath}/config/database.yml
-        ln -fs ${pkgs.writeText "secrets.yml" secretsYml} ${cfg.statePath}/config/secrets.yml
+        cp -f  ${pkgs.writeText "database.yml" databaseYml} ${cfg.statePath}/config/database.yml
+        cp -f  ${pkgs.writeText "secrets.yml" secretsYml} ${cfg.statePath}/config/secrets.yml
         ln -fs ${pkgs.writeText "unicorn.rb" unicornConfig} ${cfg.statePath}/config/unicorn.rb
 
         chown -R ${cfg.user}:${cfg.group} ${cfg.statePath}/
         chmod -R ug+rwX,o-rwx+X ${cfg.statePath}/
 
+        DBPASSWORD=$(cat ${cfg.databasePassword})
```

Comment:
Should probably be quoted to prevent whitespace in the password from breaking.

Reply:
Actually never mind, this seems to be fine.

Reply:
This quoting is needed for databasePassword to tolerate whitespace: ```DBPASSWORD=$(cat "${cfg.databasePassword}")```. The "outer" quoting is unneeded, as mentioned, although it looks a bit scary :-)

Reply:
Although while putting spaces in passwords is a sensible thing to do IMHO, anyone who's putting a space in the path to their password should reevaluate their life choices. Again IMHO ;)

Reply:
Hehe :-)

Reply:
This at least can be trivially fixed, and I'll do so. Thanks!